x/reward/keeper/msg_set_reward.go

```diff
@@ -57,25 +57,31 @@ func (k msgServer) SetRewards(goCtx context.Context, msg *types.MsgSetRewards) (
 		}
 	}
 
-	if msg.Coins.Empty() || msg.LastRewardHeight == 0 {
+	if (msg.Coins.Empty() || msg.LastRewardHeight == 0) && found {
```

Comment:
```suggestion
	if (msg.Coins.Empty() || msg.LastRewardHeight == 0) {
```

Previous behavior is right
If the pool doesn't exist and the user wants to remove it, nothing will happen because `RemoveRewardPool` is idempotent
Maybe an optimization is to make the message fails if `msg.Coins.Empty() || msg.LastRewardHeight == 0` and the pool doesn't exist

However adding `&& found` in this condition change the behavior because the reward pool would be then created with empty coins or last reward height set to 0

Reply:
> Maybe an optimization is to make the message fails if msg.Coins.Empty() || msg.LastRewardHeight == 0 and the pool doesn't exist

Actually that could make sense to do this if we now emit an event

Reply:
I think its okay just remove this extra check I added.  I removed the `EventRewardPoolUpdated` as it is unecessary